src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -29,66 +30,116 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
+            var listOfParametersOrdinals = new List<int>();
 
-            // Look for an existing "if (p == null)" statement, or "p ?? throw" check.  If we already
-            // have one, we don't want to offer to generate a new null check.
-            //
-            // Note: we only check the top level statements of the block.  I think that's sufficient
-            // as this will catch the 90% case, while not being that bad an experience even when 
-            // people do strange things in their constructors.
-            if (blockStatementOpt != null)
+            foreach (var parameterNode in listOfParameterNodes)
             {
-                if (!CanOffer(blockStatementOpt.Syntax))
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
+                if (ParameterValidForNullCheck(document, parameter, semanticModel, blockStatementOpt, cancellationToken))
                 {
-                    return ImmutableArray<CodeAction>.Empty;
+                    listOfParametersOrdinals.Add(parameter.Ordinal);
                 }
+            }
 
-                foreach (var statement in blockStatementOpt.Operations)
+            if (!listOfParametersOrdinals.Any())
+            {
+                return ImmutableArray<CodeAction>.Empty;
+            }
+
+            // Great.  The list has parameters that need null checks. Offer to add null checks for all.
+            return ImmutableArray.Create<CodeAction>(new MyCodeAction(
+                 FeaturesResources.Add_null_checks_for_all_parameters,
+                     c => DocumentRetriever(document, functionDeclaration, blockStatementOpt, listOfParametersOrdinals, c)));
+
+            // DocumentRetriever updates all necessary variables per each parameter and calls for AddNullCheckAsync
+            async Task<Document> DocumentRetriever(
+                Document document, SyntaxNode functionDeclaration, IBlockOperation blockStatementOpt, List<int> listOfParametersOrdinals, CancellationToken cancellationToken)
+            {
+                foreach (var index in listOfParametersOrdinals)
                 {
-                    if (IsIfNullCheck(statement, parameter))
+
+                    // Update functionDeclaration and use it to get the first valid ParameterNode using the ordinals (index).
+                    // var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                    var token = root.FindToken(position);
+                    var firstparameterNode = GetParameterNode(token, position);
+                    functionDeclaration = firstparameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
```

Comment:
There's already identical code in `AbstractInitializeParameterCodeRefactoringProvider`. So maybe expose a new protected helper method for this, instead of making `GetParameterNode` protected?